code/game/gamemodes/gang/vigilante.dm

```diff
@@ -0,0 +1,203 @@
+				toAdd = "<a href='?src=\ref[src];purchase=[G.id]'>[toAdd]</a>"
+			dat += toAdd
+			var/extra = G.get_extra_info(user, gang, src)
+			if(extra)
+				dat += "<br><i>[extra]</i>"
+			dat += "<br>"
+		dat += "<br>"
+
+	dat += "<a href='?src=\ref[src];choice=refresh'>Refresh</a><br>"
+
+	var/datum/browser/popup = new(user, "gangtool", "Welcome to Vigilante's Companion v1.1", 400, 750)
+	popup.set_content(dat)
+	popup.open()
+
+
+
+/obj/item/device/vigilante_tool/Topic(href, href_list)
+	if(href_list["purchase"])
+		var/datum/gang_item/G = vig_item_list[href_list["purchase"]]
+		if(G && G.can_buy(usr, gang, src))
+			G.purchase(usr, gang, src, FALSE)
+	if(href_list["destroy"])
+		Destroy_Contraband(usr)
+	attack_self(usr)
+
+/obj/item/device/vigilante_tool/proc/Destroy_Contraband(mob/living/user)
+	var/obj/item/I = user.get_active_held_item()
+	var/value
+	if(!I)
+		to_chat(user, "<span class='notice'> No item detected")
+		return
+	switch(I.type)
+		if(/obj/item/weapon/gun/ballistic/automatic/pistol)
+			value = 20
+		if(/obj/item/weapon/implanter/gang)
+			value = 12
+		if(/obj/item/weapon/reagent_containers/syringe/stimulants)
+			var/obj/item/weapon/reagent_containers/syringe/stimulants/S
+			if(S.list_reagents.len)
+				value = 10
+			else
+				value = 2
+		if(/obj/item/weapon/grenade/plastic/c4)
+			value = 5
+		if(/obj/item/toy/crayon/spraycan/gang)
+			var/obj/item/toy/crayon/spraycan/gang/SC = I
+			value = 1 + round(SC.charges/5)
+		if(/obj/item/weapon/grenade/syndieminibomb/concussion/frag)
+			value = 15
+		if(/obj/item/clothing/shoes/combat/gang)
+			value = 10
+		if(/obj/item/weapon/pen/gang)
+			value = 20
+		if(/obj/item/weapon/reviver)
+			value = 25
+		if(/obj/item/device/gangtool)
+			value = 15
+		if(/obj/item/clothing/glasses/hud/security/chameleon)
+			value = 5
+		if(/obj/item/weapon/gun/ballistic/automatic/mini_uzi)
+			value = 40
+		if(/obj/item/weapon/gun/ballistic/automatic/sniper_rifle)
+			value = 30
+		if(/obj/item/ammo_box/magazine/sniper_rounds)
+			value = 5
+		if(/obj/item/weapon/gun/ballistic/shotgun/lethal)
+			value = 25
+		if(/obj/item/weapon/gun/ballistic/automatic/surplus/gang)
+			value = 8
+		if(/obj/item/weapon/throwing_star)
+			value = 2
+		if(/obj/item/weapon/switchblade)
+			value = 5
+		if(/obj/item/weapon/storage/belt/military/gang)
+			value = 10
+		if(/obj/item/clothing/gloves/gang)
+			value = 8
+		if(/obj/item/clothing/neck/necklace/dope)
+			value = 6
+		if(/obj/item/clothing/shoes/gang)
+			value = 17
+		if(/obj/item/clothing/mask/gskull)
+			value = 15
+		if(/obj/item/clothing/head/collectable/petehat/gang)
+			value = 12
+	if(istype(I, /obj/item/clothing))
+		for(var/datum/gang/G in SSticker.mode.gangs)
+			if(I.type == G.outer_outfit && I.armor["bullet"]>=35)
+				value = 5
+	if(!value)
+		to_chat(user, "<span class='notice'> No contraband detected")
+		return
+	playsound(get_turf(loc), 'sound/items/poster_being_created.ogg', 75, 1)
+	if(do_after(usr, 30, target = user))
+		if(I)
```

Comment:
Yes but then it won't show the action bar (ive tested it). The in-hand action bars aren't working right now either.